src/components/Layout/Header/NavBar/ChainMenu.tsx

```diff
@@ -0,0 +1,121 @@
+import { ChevronDownIcon } from '@chakra-ui/icons'
+import { Menu, MenuButton, MenuGroup, MenuItem, MenuList } from '@chakra-ui/menu'
+import { Box, Button, Flex, useColorModeValue } from '@chakra-ui/react'
+import { CHAIN_NAMESPACE, fromChainId } from '@shapeshiftoss/caip'
+import { supportsEthSwitchChain } from '@shapeshiftoss/hdwallet-core'
+import { bnOrZero } from '@shapeshiftoss/investor-foxy'
+import { useEffect, useMemo, useState } from 'react'
+import { AssetIcon } from 'components/AssetIcon'
+import { CircleIcon } from 'components/Icons/Circle'
+import { getChainAdapters } from 'context/PluginProvider/PluginProvider'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { selectAssetById } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+const getSupportedEvmChainIds = () =>
```

Comment:
This is a getter function vs. a static export, because of the way JS exports work - this would give us an empty chain-adapters array, see https://262.ecma-international.org/6.0/#sec-moduleevaluation

Tried to make this a memoized expression with `featureFlags` as a dependency, which didn't work either.